src/wwwroot/js/genpage/gentab/currentimagehandler.js

```diff
@@ -237,6 +237,13 @@ function toggleShowLoadSpinners() {
 }
 
 function clickImageInBatch(div) {
+    // Remove 'image-block-selected' from all image-blocks in the batch
+    document.querySelectorAll('#current_image_batch .image-block-selected').forEach(function (block) {
```

Comment:
Please use standard `foreach` not functional forEach.
ie, `foreach (let block of document.queryetc)`

Reply:
Also this should probably be current_image_batch querySelectorAll, not document query that then only matches within current_image_batch